### PR TITLE
Check for invalid hierarchies in the asset container and warn the user

### LINKED ISSUE #13477: Treat cases in the AssetContainer where only part of the hierarchy is moved to it
Currently, if we have a mesh hierarchy and add only part of it to an AssetContainer there is some strangeness. For example, in this PG: https://playground.babylonjs.com/#5NFRVE#168 
Here the cylinder is parented to the box which is parented to the sphere, but only the cylinder and sphere are added to the container. When we call instantiateModelsToScene all the hierarchy is instantiated, but if we call removeAllFromScene, only the cylinder and sphere are removed, and we are left with a box with an invalid parent reference.
image

### PR DESCRIPTION
**Pull request by carolhmj**
Fix #13477

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/assetContainer.ts [hunk 1]
```diff
@@ -15,6 +15,9 @@ import type { Node } from "./node";
 import type { Observer } from "./Misc/observable";
 import type { ThinEngine } from "./Engines/thinEngine";
 import { InstancedMesh } from "./Meshes/instancedMesh";
+import { Light } from "./Lights";
+import { Camera } from "./Cameras";
+import { Tools } from "./Misc";
```
Comment (Popov72) [inline packages/dev/core/src/assetContainer.ts:20]:
```typescript
import { Light } from "./Lights";
import { Camera } from "./Cameras";
import { Tools } from "./Misc";
```
Shouldn't it be:
```typescript
import { Light } from "./Lights/light";
import { Camera } from "./Cameras/camera";
import { Tools } from "./Misc/tools";
```
?

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/assetContainer.ts (hunk 1 above)
Comment (sebavan) [reply packages/dev/core/src/assetContainer.ts:20]: yup I think it is why the build is failing :-)

Comment (carolhmj) [reply packages/dev/core/src/assetContainer.ts:20]: Oh dang auto imports! Lemme fix this

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13490/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13490/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13490/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13490/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13490/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13490/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13490/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13490/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13490/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by aWeirdo** [comment]
I think this broke something when using AssetContainer.instantiateModelsToScene()
Started getting spammed with logger messages "Instances should only be created for meshes with geometry."
All meshes and rootNodes added by *instantiateModelsToScene* now tagged as "isAnInstance"

    scene.meshes.forEach( (mesh) => {
        console.error(mesh.name, mesh.isAnInstance)
    })
    
check console (after hard reload), should show the logger message spam.
https://playground.babylonjs.com/#S7E00P

**Pull request merged**